aikau/src/main/resources/alfresco/html/Image.js

```diff
@@ -0,0 +1,421 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * This module provides a simple mechanism for creating image elements on a page. Additionally, it mixes
+ * in the [_PublishOrLinkMixin]{@link module:alfresco/core/_PublishOrLinkMixin} class to provide support
+ * for clicking to navigate or to publish a topic.
+ *
+ * @module alfresco/html/Image
+ * @extends external:dijit/_WidgetBase
+ * @mixes external:dojo/_TemplatedMixin
+ * @author Martin Doyle
+ */
+define([
+      "alfresco/core/_PublishOrLinkMixin",
+      "alfresco/enums/urlTypes",
+      "dijit/_TemplatedMixin",
+      "dijit/_WidgetBase",
+      "dojo/_base/declare",
+      "dojo/_base/lang",
+      "dojo/Deferred",
+      "dojo/dom-class",
+      "dojo/dom-style",
+      "dojo/text!./templates/Image.html",
+      "service/constants/Default"
+   ],
+   function(_PublishOrLinkMixin, urlTypes, _TemplatedMixin, _WidgetBase, declare, lang, Deferred, domClass, domStyle, template, AlfConstants) {
+
+      return declare([_WidgetBase, _TemplatedMixin, _PublishOrLinkMixin], {
+
+         /**
+          * @instance
+          * @typedef {object} Dimensions
+          * @property {number} width The width property
+          * @property {number} height The height property
+          */
+
+         /**
+          * An array of the CSS files to use with this widget.
+          * 
+          * @instance
+          * @type {object[]}
+          * @default [{cssFile:"./css/Image.css"}]
+          */
+         cssRequirements: [{
+            cssFile: "./css/Image.css"
+         }],
+
+         /**
+          * The HTML template to use for the widget.
+          * 
+          * @instance
+          * @override
+          * @type {string}
+          */
+         templateString: template,
+
+         /**
+          * Any optional classes to be added to the wrapped image element.
+          *
+          * @instance
+          * @type {string|string[]}
+          * @default
+          */
+         classes: null,
+
+         /**
+          * Alt text for the image.
+          *
+          * @instance
+          * @type {string}
+          * @default
+          */
+         altText: null,
+
+         /**
+          * If we can get no other sizes, then rather than having a zero-by-zero image,
+          * make it square using this value (treated as pixels).
+          *
+          * @instance
+          * @type {number}
+          * @default
+          */
+         defaultMinimumSize: 50,
+
+         /**
+          * An optional CSS height to apply to the image node (e.g. "10px" or "50%")
+          *
+          * @instance
+          * @type {string}
+          * @default
+          */
+         height: null,
```

Comment:
If we're considering eventually making Thumbnail extend this module then we want to make sure that we use consistent variable names. So Thumbnail just uses "width" as a shorthand for both "width" and "height" and then "dimensions" for the longhand form. We want to avoid inconsistencies in naming, Thumbnail is already out in the wild (and has been for a while) so can't be changed.